src/core/client.ts

```diff
@@ -315,13 +325,14 @@ export class Client<P = Indexable> implements Observable<ClientEvent<P>> {
   private key: string;
   private presenceInfo: PresenceInfo<P>;
   private status: ClientStatus;
-  private attachmentMap: Map<string, Attachment<P>>;
+  private attachmentMap: Map<DocumentKey, Attachment<P>>;
   private syncLoopDuration: number;
   private reconnectStreamDelay: number;
 
   private rpcClient: RPCClient;
   private watchLoopTimerID?: ReturnType<typeof setTimeout>;
   private remoteChangeEventStream?: any;
+  private docKeysConnectedStream: Array<string>;
```

Comment:
It looks like we can create a `docKeysConnectedStream` from `attachmentMap`, is there a reason for keeping a separate state, `docKeysConnectedStream`?

Reply:
I needed the variable `docKeysConnectedStream` to indicate which stream was unwatched in the log after cancellation, because the 'attachmentMap' would have been changed after the cancellation.

```ts
private disconnectWatchStream() {
    this.remoteChangeEventStream.cancel();
    logger.debug(
      `[WD] c:"${this.getKey()}" unwatches d:"${this.docKeysConnectedStream}"`,
    );
```

Reply:
I'm wondering if it's possible to remove docKeysConnectedStream, because the more states it has, the more complex.

Reply:
I checked again and found that `docKeysConnectedStream` is no longer needed. 
It was needed to identify the stream because the order of the `watched` and `unwatched` events was not guaranteed.
However, now that the `unwatched` event always occurs before any new `watched` events, 
it is sufficient to show which documents are connected in the `watched` event log .

![image](https://user-images.githubusercontent.com/81357083/223034879-b85058ec-b3fe-4201-b3b3-28eb7a6a16fa.png)
(The event order has been modified to be guaranteed: https://github.com/yorkie-team/yorkie/pull/479)